src/accommodation/accommodation.service.ts

```diff
@@ -199,19 +202,35 @@ export class AccommodationService {
     }
   }
 
-  async getUserAccommodations(ownerId: string) {
-    let accommodation;
+  async getUserAccommodations(ownerId: string, options: GetUserAccommodationsDto) {
     try {
-      accommodation = await this.prisma.accommodation.findMany({
+      const { page, limit, includeDeleted } = options;
+      const findAccommodationsQueryObj: any = {
         where: { ownerId },
         include: {
           address: true,
         },
-      });
+      };
+
+      if (page && limit) {
+        findAccommodationsQueryObj.skip = (+page - 1) * +limit;
+        findAccommodationsQueryObj.take = +limit;
+      }
+
+      if (includeDeleted === true) {
```

Comment:
too complex, if (includeDeleted) is enough. You already have @IsBoolean() validation in Dto

Reply:
i am doing this, cause if it is true, i will take all deleted accommodations, if false i will take all non deleted, and if it is not provided i will show all of them(deleted and non-deleted at once)